### ISSUE TITLE
Google Format plugin 1.16.0.2 does not work well with IntelliJ 2023.2

### ISSUE THREAD

**Issue by Martin-Entwistle-i2** [google/google-java-format#952]
I updated IntelliJ to 2023.2 and noticed immediately that invoking the Google formatter appears to do nothing.
If I hold donwn CTRL+ALT+L it does sometimes format the code, but most of the time is doesn't

Rolling back IntelliJ to 2023.1.5 fixes it, so it appears to be a problem specifically with the latest IntelliJ 2023.2

I did notice the popup:
![image](https://github.com/google/google-java-format/assets/97091746/cee172e9-e571-4607-9d84-784d2b1ca972)

When the code reformat was unsuccessful.  I could not see anything relevant in the IntelliJ logs

**Comment by robert-jf-close** [comment]
I experienced the same issue. I found that if I highlight all the text in a file and then ctrl+alt+L it will format. This is obviously not efficient.

**Comment by benyamynbrkyc** [comment]
I have the same issue, just hitting Cmd + Option + L will sometimes work and sometimes won't and highlighting the entire file and hitting the shortcut will format. However, sometimes that will hang and show the loading bar.

**Comment by robert-jf-close** [comment]
I have been formatting every file twice. First, ctrl + alt + l will fix the imports, then select all the text and ctrl + alt + l again to format the lines.

**Comment by simonolander** [comment]
I'm experiencing the same issue, and I can confirm that rolling back to 2023.1.5 solved it.

**Comment by Martin-Entwistle-i2** [comment]
> I'm experiencing the same issue, and I can confirm that rolling back to 2023.1.5 solved it.

Yes, rolling back IntelliJ solved this particular issue.  This is obviously a little dissapointing though as I saw some new IntelliJ features I wanted to use.

**Comment by n-n-nebbl** [comment]
Getting the same issue if i normally try to format.
It is always working if i use [ ctrl + shift + alt + l ] to open the settings popup for formatting and just hit enter without changing anything.

**Comment by ze** [comment]
Has there been any RCA? My first guess would be that the plugin does not retain a lock of the file, because I see that it briefly formats, then reverts. This leads me to assume changes have not been committed reliably, could not be committed, or were rolled back.

**Comment by Martin-Entwistle-i2** [comment]
> Following this from the README fixed it for me:
> 
> https://github.com/google/google-java-format#intellij-jre-config
> 
> Thanks to my colleague [iamr0b0tx](https://github.com/iamr0b0tx/iamr0b0tx) for pointing this out to me 👍.
> 
> The instructions changed relatively recently: [4f5ffd0](https://github.com/google/google-java-format/commit/4f5ffd0660f9aa66c1ebc8deef55802a809d3bcc) and I didn't bother to check.

Is that on `IntelliJ IDEA 2023.2.1` ? 
I had previously added those entries to the custom JVM options, not having them in recent versions of IntelliJ causes problems for sure, but even with those in the latest Intellij, it is broken for me (the only reliable way to format is to select all first)

**Comment by lmichelbacher** [comment]
That's on `2023.2.1`, yes.

**Comment by n-n-nebbl** [comment]
The issue is only there if you have optimize import also activated while formatting the code.
If i deactivate optimize imports (CTRL + SHIFT + ALT + L to open the modal) it works always so i can now use it separately (formatting with L and optimizing imports with O)

**Comment by Phanquan** [comment]
The plugin has not been working properly since the new updates on both my macOS and window, Sometimes worked, but sometimes didn't. The walk-around fix provided by @n-n-nebbl makes the work done though

**Comment by plumpy** [comment]
I think the updates in version 1.17.0.0 will hopefully fix these issues, although they're a little vague so it's hard to know for sure (thanks to https://github.com/google/google-java-format/commit/28b199c54957bb10d203bb5d7a0e3ed82603f5e8 by @facboy)

Give it a shot and let me know if there are still problems. If you see a stacktrace when things don't work, that would be the most useful (but I understand that may not always be the case).

**Comment by n-n-nebbl** [comment]
With 1.17.0.0 the code formatting seems to work again if you also activate optimize imports.
But the imports are only getting optimized if there is no formatting to do. So after two times STRG + ALT + L both is correct. First time only formatting is happening, second time the imports getting optimized.

**Comment by benyamynbrkyc** [comment]
Things seem fine with 1.17
